### PR TITLE
Fixed size oracle

### PR DESCRIPTION
**Pull request by gavraz**
Added package with the fixed size oracle

### CODE REVIEW

File: eligibility/fixedoracle.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		delete(fo.honest, client)
+	} else {
+		delete(fo.faulty, client)
+	}
+	fo.mutex.Unlock()
+}
+
+func cloneMap(m map[string]struct{}) map[string]struct{} {
+	c := make(map[string]struct{}, len(m))
+	for k, v := range m {
```
Comment (antonlerner) [inline eligibility/fixedoracle.go:58]: you iterate a map without locking it

File: eligibility/fixedoracle.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		c[k] = v
+	}
+
+	return c
+}
+
+func pickUnique(pickCount int, orig map[string]struct{}, dest map[string]struct{}) {
+	i := 0
+	for k := range orig { // randomly pass on clients
```
Comment (antonlerner) [inline eligibility/fixedoracle.go:67]: map is not locked

**Review by antonlerner — APPROVED** [review]

File: eligibility/fixedoracle.go (hunk 2 above)
Comment (gavraz) [reply eligibility/fixedoracle.go:67]: pickUnique is private and not thread-safe. We never call this function without enforcing safety.

File: eligibility/fixedoracle.go (hunk 1 above)
Comment (gavraz) [reply eligibility/fixedoracle.go:58]: cloneMap is private and not thread-safe. We never call this function without enforcing safety.

**Pull request merged**